Master updates

rashidfawad opened the pull request:
I have updated Symfony from 2.3 to 2.8 along with some other dependencies. I have also removed some deprecations. The November Camp page route did not have any Controller specified.  Perhaps we can fix it after you review the changes. 

Cheers

rashidfawad committed:
Added claer cache script

rashidfawad committed:
upgraded symfony from 2.3 to 2.8

rashidfawad committed:
removed deprecations, updated to symfony 2.8

Nyholm reviewed (changes requested):
Awesome. Thank you. Ive reviewed this PR and tested it. I like it. I only had a few comments. 

A question, Why did you change all the file permissions?

rashidfawad committed:
replaced tilde with caret, added twig extensions

rashidfawad commented:
Than you for review. I have added Twig/Extensions, replaced tilde with caret in composer. I could use FrameworkBundle TemplateAction for rendering the november-camp page but then I did not have the template so I skipped it. As each page is built using small text portions.
Regarding the file permissions usually when I need to change multiple files, I usually do "sudo chmod -R 777 abc" to make it editable :).

Nyholm commented:
Thank you. 
If you revert the file permissions, I'll be happy to merge. We do not need the world to be able to edit the source.

rashidfawad committed:
changed mode to 644

rashidfawad commented:
Hi, I can't change the permissions to 644 because when I change it I can no longer use git to commit the changes. I also could not use symfony.se the first time I cloned it from Github. I had to change the permission then I could access the site on localhost. 

If you can change the permissions, kindly do it and then merge it. Thank you.

Nyholm commented on composer.json:
```diff
@@ -4,49 +4,58 @@
+        "symfony/symfony": "2.8.*",
+        "doctrine/orm": "^2.4.8",
+        "doctrine/doctrine-bundle": "^1.4",
+        "twig/twig": "^1.0||^2.0",
```
Why do we want to support both?
Also, it fails under twig 2.0

rashidfawad replied:
No, we don't have to support both. We should update it to the latest if we want to switch to Symfony3.  If there is some deprecation warning or BC break by using version 1.0. I can remove it. It was working for me on localhost.

Nyholm commented on composer.json:
```diff
@@ -4,49 +4,58 @@
+        "knplabs/knp-markdown-bundle": "^1.5.1",
+        "dms/meetup-api-bundle": "^1.0.0",
+        "happyr/slugify-bundle": "^1.0.1",
+        "debril/rss-atom-bundle": "^3.0.5",
```
What were the BC breaks between 1.3.2 and ^3.0.5? I can see some that has not been taken care of.

Nyholm reviewed (commented):
There were some issues and fixed that has not been completed here. 

Ie, the test code did not run. I have fixed those and created a new PR.

Nyholm commented:
Ive merged the changes. Thank you for this PR.

rashidfawad commented:
I can't access the site. Can you check the logs on the server where the problem is? It's giving me 500 error.

Nyholm commented:
FIxed now. I added PHP7 and HTTPS

rashidfawad commented:
Yuup, saw it in the commit. The error may have occurred because we were referencing Http links from Https.
